lib/src/components/NxFileUpload/NxMultiFileUpload/NxMultiFileUpload.tsx

```diff
@@ -169,6 +170,9 @@ const NxMultiFileUpload = forwardRef<HTMLDivElement, Props>(function NxMultiFile
                   className="nx-file-upload__select-btn">
           Add Files
         </NxButton>
+        <span id={totalFilesSelectedId} className="nx-multi-file-upload__file-count">
+          {isFileSelected ? `${files.length} files selected` : 'No File Selected'}
```

Comment:
You've now got "No File Selected" in two different places (with inconsistent capitalization, though that's not important).  You should be able to merge those two elements (and possibly also use the same element to handle the "___ files selected" text).

Reply:
> You've now got "No File Selected" in two different places (with inconsistent capitalization, though that's not important). You should be able to merge those two elements (and possibly also use the same element to handle the "___ files selected" text).

Added a variable to track number of files added (if any): [055bbb2](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/055bbb2a8e59bb3c6ca4ce3cd5f5de9eef7d22a0)